### PR TITLE
Optional C++ and TLS tests

### PR DESCRIPTION
**Pull request by jforissier**
Addressing https://github.com/OP-TEE/optee_test/issues/458.

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jforissier** [comment]
I have `s/CPLUSPLUS/CXX/` in the second patch (see https://github.com/OP-TEE/build/pull/441). Tag applied to first patch only.

File: ta/os_test/sub.mk [hunk 1]
```diff
@@ -8,7 +10,7 @@ srcs-y += init.c
 srcs-y += os_test.c
 srcs-y += ta_entry.c
 srcs-$(CFG_TA_FLOAT_SUPPORT) += test_float_subj.c
-ifneq ($(COMPILER),clang)
+ifeq ($(WITH_CXX_TESTS),1)
```
Comment (etienne-lms) [inline ta/os_test/sub.mk:13]: s/1/y/ ?

**Review by etienne-lms — COMMENTED** [review]
In commit long for 1st commit "Introduce WITH_TLS_TESTS": missing `[1]` in  `Link:`.

+ comment in 2nd commit.

File: ta/os_test/sub.mk (hunk 1 above)
Comment (jforissier) [reply ta/os_test/sub.mk:13]: Good catch, fixed and tested properly with both `WITH_CXX_TESTS` set to `y` and `n` this time.

**Comment by jforissier** [comment]
@jenswi-linaro may I apply your R-b tag to "Introduce WITH_CXX_TESTS"?

**Comment by jenswi-linaro** [comment]
@jforissier, sure. Looks good.

**Comment by jforissier** [comment]
IBART reports a failure but the log looks OK, although it appears to be truncated. @jbech-linaro did the test time out perhaps? Should I just try to restart the test?

**Pull request merged**